### ISSUE TITLE
Git clone or submodule update does not find private key file if depth flag is set

### ISSUE THREAD

**Issue by idemax** [git-for-windows/git#2508]
I'm cloning a repository with a few submodules. In order to save time I'm doing a shallow clone passing the private key as environment variable. Example:

    GIT_SSH_COMMAND="ssh -i temp_pvt_key.key" git clone -b my_branch --depth=1 git@my-repo.com:my_team/my_repo.git temp_repo

The command above works just fine! But when I try do the same with submodules adding the `--depth=1`, git is not able to find the `temp_pvt_key.key` anymore.

**Error:**

    Cloning into 'C:/path/to/submodule/folder'...
    Warning: Identity file ../temp_pvt_key.key not accessible: No such file or directory.
    git@my-repo.com: Permission denied (publickey).
    fatal: Could not read from remote repository.

*I'm using relative path (`../`) once the private key file is in the repository parent folder.*

**Works but the whole submodule repository is cloned:**

    cd temp_repo
    GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ../temp_pvt_key.key" git submodule update --init

**Basically the same command plus --depth=1. This does not work at all:**

    cd temp_repo
    GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ../temp_pvt_key.key" git submodule update --depth=1 --init

**I've tried in a single command too but no success, I ended up with the same error:**

    GIT_SSH_COMMAND="ssh -i temp_pvt_key.key" git clone -b my_branch --depth=1 --recursive --shallow-submodules git@my-repo.com:my_team/my_repo.git temp_repo

*Git version: `git version 2.19.2.windows.1`*

**Comment by dscho** [comment]
> when I try do the same with submodules adding the `--depth=1`, git is not able to find the `temp_pvt_key.key` anymore.

Is the reason maybe that you specified a relative path, and that this path is interpreted relative to the current working directory (which _must_ be different for submodules)?

In other words, this should work (and is the correct way anyway):

```sh
GIT_SSH_COMMAND="ssh -i \"$PWD/temp_pvt_key.key\"" \
git clone -b my_branch --depth=1 git@my-repo.com:my_team/my_repo.git temp_repo
```

**Comment by idemax** [comment]
> > when I try do the same with submodules adding the `--depth=1`, git is not able to find the `temp_pvt_key.key` anymore.
> 
> Is the reason maybe that you specified a relative path, and that this path is interpreted relative to the current working directory (which _must_ be different for submodules)?
> 
> In other words, this should work (and is the correct way anyway):
> 
> ```shell
> GIT_SSH_COMMAND="ssh -i \"$PWD/temp_pvt_key.key\"" \
> git clone -b my_branch --depth=1 git@my-repo.com:my_team/my_repo.git temp_repo
> ```

Thanks @dscho but I tried that approach either and have no success.

This code example that you mentioned works pretty fine, the problem happens when you try this updating the submodules.

**Comment by dscho** [comment]
@idemax do you have any verbose output handy, after setting `GIT_TRACE=1`?
